dotcom-rendering/src/web/lib/useStickyObserver.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, useRef } from 'react';
+
+/**
+ * Uses an intersection observer to determine if
+ * @param {IntersectionObserverInit} options - Intersection observer options
+ * */
+const useStickyObserver = (
```

Comment:
The difference between the `useStickyObserver` hook I created and `useHasBeenSeen` is that `useStickyObserver` resets the `isIntersecting` whereas with `useHasBeenSeen` my understanding is that it is only ever set to `true` once, and thereafter stays true. This means once the video has been stuck it stays stuck until we click it closed, after which it stays closed. Essentially the hook only updates the state once.

We could set `isIntersecting` to `false` once it's no longer intersecting, but I hadn't wanted to change it in case that went against the initial intention for that hook. I think it's probably the case, but it's worth confirming with @benwuersching that we want to be able to repeatedly stick and unstick videos. 

This is the current behaviour if we use `useHasBeenSeen`:

https://user-images.githubusercontent.com/77005274/155010806-626213b0-c6e9-47ca-b8d8-8869881e1e17.mov

Reply:
Thanks for the (really useful!) explanation. 

🤔 it seems like this is kind of a naming problem. I'd be wary of creating an option on `useHasBeenSeen` that was named something specific to stickiness but perhaps the option can be named something related to what it is doing? 

Maybe `options.repeat` or `options.resetOnExit` or `options.dynamic` or `options.fireWhenHidden` or `options.fireOnExit`, etc.

Reply:
Yeah that makes sense, I think `repeat` wins it for me. Have added to `useHasBeenSeen` and removed the `useStickyObserver` hook.